source/common/upstream/load_balancer_impl.cc

```diff
@@ -47,12 +47,29 @@ LoadBalancerBase::LoadBalancerBase(const PrioritySet& priority_set, ClusterStats
     recalculatePerPriorityState(host_set->priority(), priority_set_, per_priority_load_,
                                 per_priority_health_);
   }
+  // Reclaculate panic mode for all levels.
+  recalculatePerPriorityPanic();
+
   priority_set_.addMemberUpdateCb([this](uint32_t priority, const HostVector&,
                                          const HostVector&) -> void {
     recalculatePerPriorityState(priority, priority_set_, per_priority_load_, per_priority_health_);
   });
+  priority_set_.addMemberUpdateCb(
+      [this](uint32_t priority, const HostVector&, const HostVector&) -> void {
+        UNREFERENCED_PARAMETER(priority);
+        recalculatePerPriorityPanic();
+      });
 }
 
+// The following cases are handled by
+// recalculatePerPriorityState and recalculatePerPriorityPanic methods;
+// - Total health is = 100. It means there are enough healthy hosts to handle the load.
```

Comment:
Can we refer to this as "normalized total health" with a description of what that means? Same below in comment?

Reply:
Done.